apps/console/src/features/roles/components/edit-role/edit-role-common/role-api-resources-list-item.tsx

```diff
@@ -82,7 +82,7 @@ export const RoleAPIResourcesListItem: FunctionComponent<RoleAPIResourcesListIte
 
         // Check if all scopes are selected from the api resource
         const isAllScopesSelected = (): boolean =>
-            apiResource?.scopes?.length === selectedPermissions?.length;
+            apiResource?.scopes?.length !== 0 && apiResource?.scopes?.length === selectedPermissions?.length;
```

Comment:
Do we really need `apiResource?.scopes?.length !== 0 && ` check here?

```suggestion
            apiResource?.scopes?.length === selectedPermissions?.length;
```

Reply:
Currently even without scopes we can check this tick, but since no scopes are selected this api resource wont be added to the role. IMO it's kinda misleading.

<img width="727" alt="image" src="https://github.com/wso2/identity-apps/assets/61885844/c9dbda0b-4497-409c-b04a-a85fe2111f10">